pkg/controllermanager/controller/shoot/shoot_reference_control.go

```diff
@@ -136,17 +146,29 @@ func (r *shootReferenceReconciler) reconcileShootReferences(c client.Client, sho
 		return err
 	}
 
+	// Iterate over all user configmaps in project namespace and check if they can be released.
+	if err := r.releaseUnreferencedConfigMaps(c, shoot); err != nil {
+		return err
+	}
+
 	// Remove finalizer from shoot in case it's being deleted and not handled by Gardener any more.
 	if shoot.DeletionTimestamp != nil && !controllerutils.HasFinalizer(shoot, gardencorev1beta1.GardenerName) {
 		return controllerutils.PatchRemoveFinalizers(r.ctx, c, shoot, FinalizerName)
 	}
 
 	// Add finalizer to referenced secrets that are not managed by Gardener.
-	needsFinalizer, err := r.handleReferencedSecrets(c, shoot)
+	addedFinalizerToSecret, err := r.handleReferencedSecrets(c, shoot)
 	if err != nil {
 		return err
 	}
 
+	addedFinalizerToConfigMap, err := r.handleReferencedConfigMap(c, shoot)
```

Comment:
```suggestion
	addedFinalizerToConfigMap, err := r.handleReferencedConfigMaps(c, shoot)
```

Reply:
Actually, no. It adds finalizer only to the config map referenced by the current shoot.